plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorUtils.java

```diff
@@ -128,6 +128,15 @@ private static void findScriptList(@NotNull DBPProject project, IFolder folder,
         return result;
     }
 
+    public static List<ResourceInfo> getScriptsFromProject(@NotNull DBPProject dbpProject) throws CoreException {
+        IFolder resourceDefaultRoot = DBWorkbench.getPlatform().getWorkspace().getResourceDefaultRoot(dbpProject, ScriptsHandlerImpl.class, false);
+        if (resourceDefaultRoot != null) {
+            return Arrays.stream(resourceDefaultRoot.members()).map(it -> (IFile) it).map(it -> new ResourceInfo(it, EditorUtils.getFileDataSource(it))).collect(Collectors.toList());
```

Comment:
No checks for `instanceof IFile`. This logic may fail.

Also, such usage of Stream API will result in two (!) synthetic methods being created in the compiled code.

It's entirely unnecessary to use streams here. It could be unrolled easily to look more readable and friendly.